iceberg-rust/src/table/transaction/append.rs

```diff
@@ -0,0 +1,84 @@
+use std::cmp::Ordering;
+
+use iceberg_rust_spec::{manifest::ManifestEntry, values::Value};
+use smallvec::{smallvec, SmallVec};
+
+use crate::{
+    error::Error,
+    util::{cmp_dist, struct_to_smallvec, sub, Rectangle},
+};
+
+/// Split sets of datafiles depending on their partition_values
+#[allow(clippy::type_complexity)]
+pub(crate) fn split_datafiles_once(
+    files: impl Iterator<Item = Result<ManifestEntry, Error>>,
+    rect: Rectangle<Value>,
+    names: &SmallVec<[&str; 4]>,
+) -> Result<[(Vec<ManifestEntry>, Rectangle<Value>); 2], Error> {
+    let mut smaller = Vec::new();
+    let mut larger = Vec::new();
+    let mut smaller_rect = None;
+    let mut larger_rect = None;
+
+    for manifest_entry in files {
+        let manifest_entry = manifest_entry?;
+        let position = struct_to_smallvec(manifest_entry.data_file().partition(), names)?;
```

Comment:
this helper is more confusing that actually inlining the extraction:
```suggestion
        let partition_struct = manifest_entry.data_file().partition();
        let position: SmallVec<Value> =  names
        .iter()
        .map(|x| partition_struct.get(x).and_then(|x| x.clone()))
        .collect::<Option<_>>()
        .ok_or(Error::InvalidFormat("Partition struct".to_owned()))?:
```

Reply:
Since it's being used multiple times across the crate I would rather put it into a function.

Reply:
ah sory, I missed that it was also used in operation.rs